Remove menu item together with all nested menus

alroniks opened the pull request:
### What does it do?
It removes all nested items when the parent menu was deleted.

### Why is it needed?
Before: when parent menu is deleted nested submenus become ghosts. It may confuse users, especially when they try to create a new item but get an error that item already exists. And it is the truth but it is impossible to manage these ghosts by UI. Now the whole branch should be deleted.

### Related issue(s)/PR(s)
#14530

Commit:
Remove menu item together with all nested menus

Ruslan-Aleev commented on core/lexicon/en/menu.inc.php:
```diff
@@ -25,7 +25,7 @@
 $_lang['menu_parent_err_nf'] = 'Parent menu not found!';
-$_lang['menu_remove'] = 'Delete Menu Item';
+$_lang['menu_remove'] = 'Delete Menu';
 $_lang['menu_top'] = 'Top Menu';
```
Maybe it makes sense to change "Top Menu" to "Main Menu"? Since in the MODX3 menu is no longer on top.

JoshuaLuckers commented on core/lexicon/en/menu.inc.php:
```diff
@@ -16,7 +16,7 @@
 $_lang['lexicon_key_desc'] = 'The text, or lexicon key, that will be used for rendering the title text of this page in the menu.';
 $_lang['menu_create'] = 'New Menu';
-$_lang['menu_confirm_remove'] = 'Are you sure you want to remove this menu?';
+$_lang['menu_confirm_remove'] = 'All nested menus also will be deleted. Are you sure you want to remove this menu? ';
```
To keep it consistent with the lexicon [`resource_delete_confirm`](https://github.com/modxcms/revolution/blob/cfa09fabf3914d6012f7c4606cf1e9e601bd95a4/core/lexicon/en/resource.inc.php#L47) I propose to use this lexicon instead:
```suggestion
$_lang['menu_confirm_remove'] = 'Are you sure you want to delete this menu?<br />NOTE: Any nested menus will also be deleted!';
```

Ruslan-Aleev replied:
Then it is better to write "to delete" instead of "to remove", as in the resource lexicon.

Mark-H commented on core/model/modx/processors/system/menu/remove.class.php:
```diff
@@ -16,20 +16,47 @@
+
+        foreach ($this->modx->getIterator($this->classKey, $criteria) as $subMenu) {
+            $this->removeNested($subMenu);
+            $this->modx->runProcessor('system/menu/remove', ['text' => $subMenu->get('text')]);
```
Is this `$this->modx->runProcessor` call needed if it's already calling `$this->removeNested()` above?

JoshuaLuckers replied:
I think `$this->modx->runProcessor` is needed but `$this->removedNested()` isn't because it's being called in `beforeRemove()`.

alroniks replied:
It's needed because of it a recursive call, `removeNested` removes only items that inside, `runProcessor` removes the current item.

alroniks replied:
@JoshuaLuckers Both needed. The first one (in `beforeRemove`) starts removing after that it runs recursively to reach all levels.

alroniks committed:
Update core/lexicon/en/menu.inc.php

Co-Authored-By: Joshua Lückers <joshualuckers@me.com>

alroniks committed:
Rename Top Menu to Main Menu 

due to that fact that in 3.x no more top menu, all were moved to the left sidebar.

Mark-H commented:
Thanks @Alroniks :)
